src/components/sidebar/SideBar.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import SideBarLink from './SideBarLink';
+import SideBarButton from './SideBarButton';
+import './Sidebar.scss';
+
+import SettingsIcon from '../../images/SettingsIcon.svg';
+import LogoutIcon from '../../images/LogoutIcon.svg';
+import { ReactComponent as ClubIcon } from '../../images/ClubIcon.svg';
+
+class SideBar extends Component {
+    render(): React.ReactNode {
+        return (
+            <div id="sidebar">
```

Comment:
I think using the built in menu functionality provided by Bulma will make the layout simpler, https://bulma.io/documentation/components/menu/

Components are still good thought, I would just wrap them in a menu-list for example